src/utils/kbn_field_types.js

```diff
@@ -0,0 +1,105 @@
+
+export class KbnFieldType {
+  constructor({ name, sortable, filterable, esTypes }) {
+    this.name = name;
+    this.sortable = !!sortable;
+    this.filterable = !!filterable;
+    this.esTypes = Object.freeze(esTypes || []);
+    Object.freeze(this);
```

Comment:
I was just playing around with `Object.freeze()` in another context and, unfortunately, I don't think it's a good idea to use it since it prevents angular from rendering it in an `ng-repeat`, which requires adding a unique `$$hashKey` to the object... I'll do something else.